<?xml version="1.0"?>
	<robot name="ur3">

		<link name="base_link">
		</link>
		
		<link name="laser">
		</link>
                
        <link name="webcam">
		</link>

		<link name="imu">
		</link>

        <!--
		<link name="camera_link">
		</link>

		<link name="kinect_pivot">
		</link>

		-->

		<link name="kinect_base">
		</link>

		<link name="kinect_rotator">
		</link>
                

		<joint name="base_link_to_laser" type="fixed">
			<parent link="base_link"/>
			<child link="laser"/>
			<origin xyz="0 0 0.3429" rpy="0.0 0.0 1.570796"/>
			<axis xyz="0 0 0"/>
		</joint>

        <joint name="base_link_to_webcam" type="fixed">
			<parent link="base_link"/>
			<child link="webcam"/>
			<origin xyz="0.21 -0.051 0.27" rpy="0.0 0.0 0"/>
			<axis xyz="0 0 0"/>
		</joint>	

		<joint name="base_link_to_imu" type="fixed">
			<parent link="base_link"/>
			<child link="imu"/>
			<origin xyz="-0.198 -0.153 0.458" rpy="0.0 0.0 3.14159"/>
			<axis xyz="0 0 0"/>
		</joint>	

		<joint name="base_link_to_kinect_rotator" type="fixed">
			<parent link="base_link"/>
			<child link="kinect_rotator"/>
			<origin xyz="0.1 0.060 1.19" rpy="0.0 0.0 0.0"/>
			<axis xyz="0 0 0"/>
		</joint>

        <!--
		<joint name="kinect_pivot_to_camera_link" type="fixed">
			<parent link="kinect_pivot"/>
			<child link="camera_link"/>
			<origin xyz="0.0272 0.0117 0.034" rpy="0.0 0.0 0.0"/>
			<axis xyz="0 0 0"/>
		</joint>	

		-->

		<joint name="kinect_rotator_to_kinect_base" type="fixed">
			<parent link="kinect_rotator"/>
			<child link="kinect_base"/>
			<!-- <origin xyz="0.0067 0 0.0169" rpy="0.0 0.0 0.0"/> -->
			<origin xyz="0.0102 0 0.0306" rpy="0.0 0.0 0.0"/>
			<axis xyz="0 0 0"/>
		</joint>	

		<!-- So kinect rotator: to before angler: 6mm forward.-->
		<!-- forward base to kinect base (with angler) 9.4mm high, 13 degree angle-->
		<!-- angler to kinect pivot: 7.7mm (strate up from angler)-->
		<!-- kinect pivot to camera: 34 mm up, 27.2mm forward, 11.7mm sideways-->

		
	</robot>
